src/spec-node/imageMetadata.ts

```diff
@@ -494,8 +494,11 @@ export function getContainerEnvMetadata(containerEnv: Record<string, string> | u
 	if (!containerEnv) {
 		return '';
 	}
-
 	const keys = Object.keys(containerEnv);
-	const concatenatedEnv = keys.map(k => `ENV ${k}=${containerEnv![k]}`).join('\n');
+	// https://docs.docker.com/engine/reference/builder/#envs
+	const concatenatedEnv = keys.map(k => `ENV ${k}="${containerEnv[k]
+		.replace(/(?=["\\$])/g, '\\') // escape " and \
```

Comment:
What's the purpose of matching the `$` character? Can you add a test to help explain and cover this scenario? Same thing for the backslash literal `\`.